### PR TITLE
reference, faq: update TiDB parameters, delete command option file

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? 

- Update wording and format in TiDB cmd-line parameters
- Apply the alphabetic order for parameters in `configuration.md` to make it easier to locate
- Delete the obsolete command option file `reference/configuration/tidb-server/server-command-option.md`
- Add more omitted parameters to `tidb-server/configuration-file.md`

### What is the related PR or file link(s)? 

N/A

### Which version does your change affect? 

dev, v3.0, v2.1

### CODE REVIEW

File: dev/reference/configuration/tidb-server/configuration.md [hunk 1]
```diff
@@ -5,142 +5,141 @@ category: reference
...
 
-## `--run-ddl`
+## `--host`
 
-+ tidb-server 是否运行 DDL 语句，集群内大于两台以上 tidb-server 时设置
-+ 默认：true
-+ 值可以为 (true) 或者 (false). (true) 表明自身会运行 DDL. (false) 表明自身不会运行 DDL
++ TiDB 服务监听的 host
++ 默认："0.0.0.0"
++ 0.0.0.0 默认会监听所有的网卡 address。如果有多块网卡，可以指定对外提供服务的网卡，如 192.168.100.113
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration.md]:
```suggestion
+ 0.0.0.0 默认会监听所有的网卡地址。如果有多块网卡，可以指定对外提供服务的网卡，如 192.168.100.113
```

File: dev/reference/configuration/tidb-server/configuration.md [hunk 2]
```diff
@@ -5,142 +5,141 @@ category: reference
...
 > **注意：**
 >
-> 请不要配置成 0，除非特殊情况，一般使用默认值即可。
+> 不要将该参数配置为 `0`。除非特殊情况，一般使用默认值即可。
 
-## `--cors`
+## `--report-status`
 
-+ 用于设置 TiDB HTTP 状态服务的 Access-Control-Allow-Origin
++ 打开 (true) 或者关闭 (false) 服务状态监听端口
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration.md]:
```suggestion
+ 打开 (值为 `true`) 或者关闭 (值为 `false`) 服务状态监听端口
```

File: dev/reference/configuration/tidb-server/configuration.md [hunk 3]
```diff
@@ -5,142 +5,141 @@ category: reference
...
++ 默认：true
++ 值可以为 (true) 或者 (false)。(true) 表明开启状态监听端口；(false) 表明关闭状态监听端口
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration.md]:
```suggestion
+ 值可以为 `true` 或者 `false`。`true` 表明开启状态监听端口；`false` 表明关闭状态监听端口
```
实际的值是 `true` 还是 `(true)`? @ericsyh

File: dev/reference/configuration/tidb-server/configuration.md [hunk 4]
```diff
@@ -5,142 +5,141 @@ category: reference
...
+
+## `--run-ddl`
+
++ tidb-server 是否运行 DDL 语句，集群内大于两台以上 tidb-server 时设置
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration.md]:
```suggestion
+ tidb-server 是否运行 DDL 语句，集群内有大于两台以上 tidb-server 时设置
```

File: dev/reference/configuration/tidb-server/configuration.md [hunk 5]
```diff
@@ -5,142 +5,141 @@ category: reference
...
++ 默认：true
++ 值可以为 (true) 或者 (false)。(true) 表明自身会运行 DDL；(false) 表明自身不会运行 DDL
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration.md]:
```suggestion
+ 值可以为 `true` 或者 `false`。`true` 表明自身会运行 DDL；`false` 表明自身不会运行 DDL
```

File: dev/reference/configuration/tidb-server/configuration.md (hunk 3 above)
Comment (lilin90) [reply dev/reference/configuration/tidb-server/configuration.md]: @yikeke 为 `true`，没有括号。See [this line](https://github.com/pingcap/tidb/blob/release-3.0/config/config.toml.example#L120).

File: dev/reference/configuration/tidb-server/configuration.md (hunk 4 above)
Comment (lilin90) [reply dev/reference/configuration/tidb-server/configuration.md]: @ericsyh 这里包括两台吗？

**Comment by lilin90** [comment]
@ericsyh PTAL

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 6]
```diff
@@ -365,7 +363,7 @@ TiDB Binlog 相关配置。
 
 + 忽略写 binlog 发生的错误时处理开关，推荐不修改该值。
 + 默认值：false
-+ "true"，发生错误时，停止写入 binlog, 并且在监控项 tidb_server_critical_error_total 上计数加1；"false": 写入 binlog 失败，会停止整个 TiDB 的服务。
++ "true"，发生错误时，停止写入 binlog，并且在监控项 `tidb_server_critical_error_total` 上计数加 1；"false": 写入 binlog 失败，会停止整个 TiDB 的服务。
```
Comment (ericsyh) [inline dev/reference/configuration/tidb-server/configuration-file.md]: ### 
```suggestion
+ "true"，发生错误时，TiDB 停止写入 binlog，并且在监控项 `tidb_server_critical_error_total` 上计数加 1；"false": 写入 binlog 失败，会停止整个 TiDB 的服务。
```

File: dev/reference/configuration/tidb-server/configuration.md [hunk 7]
```diff
@@ -5,142 +5,141 @@ category: reference
...
-+ TiDB 服务会监听这个 host
-+ 0.0.0.0 默认会监听所有的网卡 address。如果有多块网卡，可以指定对外提供服务的网卡，譬如192.168.100.113
-
-## `-P`
+## `--binlog-socket`
 
-+ TiDB 服务监听端口
-+ 默认："4000"
-+ TiDB 服务将会使用这个端口接受 MySQL 客户端发过来的请求
++ TiDB 服务使用 unix socket file 方式接受内部连接，如 PUMP 服务
```
Comment (ericsyh) [inline dev/reference/configuration/tidb-server/configuration.md]:
```suggestion
+ TiDB 服务使用 unix socket file 方式接受内部连接，如 Pump 服务
```

**Comment by lilin90** [comment]
@ericsyh Addressed comments. PTAL again~ Thanks!

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 8]
```diff
@@ -214,7 +212,7 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 ### `cross-join`
 
 + 默认值：true
-+ 在做 join 的时候，两边表没有任何条件（where 字段），默认值可以执行这样的语句。但是设置为 false，则如有这样的 join 语句出现，server 会拒绝执行
++ 如果设置为 true，在做 join 时即便两边表没有任何条件（where 字段），默认可以执行这样的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```
Comment (ericsyh) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 默认在做 join 时即便两边表没有任何条件（where 字段），可以执行这样的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```

File: v2.1/reference/configuration/tidb-server/configuration-file.md [hunk 9]
```diff
@@ -123,54 +145,68 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
-+ 默认：10
-+ 如果 TiDB 超过 `retry-limit` 次重试还是没有成功，将会返回错误。
++ 默认值：false
 
 ### `cross-join`
 
-+ 默认：true
-+ 在做 join 的时候，两边表没有任何条件（where 字段），默认可以执行这样的语句。但是设置为 false，则如有这样的 join 语句出现，server 会拒绝执行
++ 默认值：true
++ 如果设置为 true，在做 join 时即便两边表没有任何条件（where 字段），默认可以执行这样的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```
Comment (ericsyh) [inline v2.1/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 如果可以执行在做 join 时即便两边表没有任何条件（where 字段）的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```

File: v3.0/reference/configuration/tidb-server/configuration-file.md [hunk 10]
```diff
@@ -215,7 +213,7 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 ### `cross-join`
 
 + 默认值：true
-+ 在做 join 的时候，两边表没有任何条件（where 字段），默认值可以执行这样的语句。但是设置为 false，则如有这样的 join 语句出现，server 会拒绝执行
++ 如果设置为 true，在做 join 时即便两边表没有任何条件（where 字段），默认可以执行这样的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```
Comment (ericsyh) [inline v3.0/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 如果可以执行在做 join 时即便两边表没有任何条件（where 字段）的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```

File: dev/reference/configuration/tidb-server/configuration-file.md (hunk 8 above)
Comment (ericsyh) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 如果可以执行在做 join 时即便两边表没有任何条件（where 字段）的语句；如果设置为 false，则有这样的 join 语句出现时，server 会拒绝执行
```

File: dev/reference/configuration/tidb-server/configuration.md (hunk 4 above)
Comment (ericsyh) [reply dev/reference/configuration/tidb-server/configuration.md]: 这个感觉描述有点问题，run-ddl 参数在一台 tidb， 两台 tidb 时候都可以设置。
只有设置 run-ddl 为 true 的 tidb 才会参数 ddl owner 的选举。所以集群里至少有一台 tidb 是设置了 run-ddl = true 的。

**Comment by ericsyh** [comment]
Rest LGTM

File: dev/reference/configuration/tidb-server/configuration.md (hunk 4 above)
Comment (lilin90) [reply dev/reference/configuration/tidb-server/configuration.md]: @ericsyh Got it. I'll update it.

**Pull request merged**